solutions/dmoj-bots.mdx

```diff
@@ -0,0 +1,201 @@
+---
+id: dmoj-bots
+source: DMOJ
+title: Bots
+author: Kevin Sheng
+---
+
+## Explanation
+
+Let us first define a DP recurrence relation, with $\texttt{dp}[b][r]$ being the number of unique states where the blue bot has made $b$ moves and the red bot has made $r$ moves:
+$$
+\texttt{dp}[b][r]=\texttt{dp}[b - 1][r] + \texttt{dp}[b][r - 1]
+$$
+It follows that the answer is the sum of all the elements in $\texttt{dp}$.
+
+This is what $\texttt{dp}$ might look like for $n=4$:
+<table className="text-center">
+    <tbody>
+        <tr>
+            <td>1</td>
+            <td>1</td>
+            <td>1</td>
+            <td>1</td>
+            <td>1</td>
+        </tr>
+        <tr>
+            <td>1</td>
+            <td>2</td>
+            <td>3</td>
+            <td>4</td>
+            <td>5</td>
+        </tr>
+        <tr>
+            <td>1</td>
+            <td>3</td>
+            <td>6</td>
+            <td>10</td>
+            <td>15</td>
+        </tr>
+        <tr>
+            <td>1</td>
+            <td>4</td>
+            <td>10</td>
+            <td>20</td>
+            <td>35</td>
+        </tr>
+        <tr>
+            <td>1</td>
+            <td>5</td>
+            <td>15</td>
+            <td>35</td>
+            <td>70</td>
+        </tr>
+    </tbody>
+</table>
+
+Note that these numbers are exactly the same as the ones in Pascal's triangle, only rotated a bit! These numbers form a rotated square of length $n+1$ (henceforth referred to as $x$), and we can express its sum as
+$$
+\sum_{i=0}^{x-1} \sum_{j=i}^{x+i-1} {j \choose i}
+$$
+Using the [hockey stick identity](https://brilliant.org/wiki/hockey-stick-identity/), we see that this equals
+$$
+\begin{align*}
+&\sum_{i=0}^{x-1} {x+i \choose i+1}\\
+=&\sum_{i=1}^{x} {x+i-1 \choose i}
+\end{align*}
+$$
+We also know from the hockey stick identity that $\sum_{i=0}^{a} {a+i-1 \choose i} = {a+2 \choose a}$. Thus, we get that
```

Comment:
```suggestion
We also know from the hockey stick identity that $\sum\limits_{i=0}^{a} {a+i-1 \choose i} = {a+2 \choose a}$. Thus, we get that
```

Reply:
wait what does this do though

Reply:
why don't you try? (it formats the limits better)